src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx

```diff
@@ -504,6 +504,22 @@ namespace xclcpuemhal2 {
     sock = new unix_socket("EMULATION_SOCKETID");
   }
 
+  void CpuemShim::getCuRangeIdx() {
+    std::string instance_name = "";
+    for (const auto& kernel : m_xclbin.get_kernels()) {
+      // get properties of each kernel object
+      auto props = xrt_core::xclbin_int::get_properties(kernel);
+      //get CU's of each kernel object.iterate over CU's to get arguments
+      if (props.address_range != 0 && !props.name.empty()) {
+        for (const auto& cu : kernel.get_cus()) {
+          instance_name = cu.get_name();
+          if (!instance_name.empty())
+            mCURangeMap[instance_name] = props.address_range;
+        }
```

Comment:
Please use early-exit (in this case early-continue) to reduce indentation.   Also, avoid the copy of kernel_properties.
```
const auto& props = ...;
if (props.address_range == 0 || props.name.empty())
  continue;

for (const auto& cu: ...) {
  ...
}
```